### DISCUSSION TITLE
Nextjs is hanging when navigating between pages

### DISCUSSION THREAD

**Discussion by salahbloovo** [vercel/next.js#24494] [Help]
Hello, 

I'm developing a new project for my company and I'm facing this weird issue. When navigating through rhe application, it hangs until resources are being installed behind the scene in- browser.

Please check my question on Stackoverflow https://stackoverflow.com/questions/67257908/nextjs-is-installing-resources-and-media-and-prevents-any-action-on-page

**Comment by wwwdepot** [comment]
Try lazy loading video resources? https://web.dev/lazy-loading-video/

**Reply by salahbloovo** [reply]
why does it get installed in the first place? and why does the application get hanged?

**Reply by ljosberinn** [reply]
why does it get loaded? thats your site, you use those assets. navigating to a new page makes those requests because you set it up to do so.
why does it hang? assets are probably so large that it just takes ages to load. you even show it in your screenshot:
![image](https://user-images.githubusercontent.com/29307652/116375031-2b568b80-a80f-11eb-8a8c-28b74cee2628.png)
10mb takes 24 seconds for your internet connection to download. consider reducing the size / quality / dimensions of the video(s).

**Reply by salahbloovo** [reply]
my friend, I've been developing apps for 15 years, what you are telling me is no sense!
- Yes, it's my app, but the client might upload a 100MB+ video on their system and they don't expect any delays in the page performance!
- If we assume that this is the correct behavior of the web (which is not) why would the app be prevented from navigating to another page because a resource that is never used/clicked on is being installed behind the scene!
- My issue is with Nextjs freezing any action on the app until all resources are installed/loaded in-browser (never experienced this issue with any other frameworks I used)
- 10MB in 23 sec to download is very slow, but this is in browser not actual download, it takes a second if I run the video.
@ljosberinn

**Comment by georgekrax** [comment]
Same here!
